contracts/carrot-app/src/handlers/execute.rs

```diff
@@ -253,8 +259,110 @@ fn withdraw(
     Ok(app_response)
 }
 
-/// Auto-compound the position with earned fees and incentives.
+fn withdraw_to_asset(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    amount: Uint128,
```

Comment:
I thought we decided that this `amount` will stay the LP amount. Vlad has already implemented the logic that was needed for this change so we should stick with that. 

This function should only swap one of the withdrawn assets into the other.

Reply:
> I thought we decided that this `amount` will stay the LP amount. Vlad has already implemented the logic that was needed for this change so we should stick with that.
> This function should only swap one of the withdrawn assets into the other.

I know that Vlad implemented logic to calculate amount of tokens from amount of liquidity, it's still liquidity amount for `withdraw` method. But I was pretty sure the idea of new method `withdraw_to_asset` was to withdraw to asset amount.

Also want to clear up if `to_asset` have to be only one of the two assets from the pool or could be any?

Reply:
`withdraw_to_asset` should take the lp token `amount` and use that amount to withdraw. The tokens received from that withdraw should then be used to swap to the `to_asset` denom. 

We could limit it to just allow it to the specific token but we might as well accept any token and rely on ANS to resolve it!

Reply:
Gotcha, thanks!